trunk/SUAVE/Components/Energy/Converters/Rotor.py

```diff
@@ -140,152 +150,220 @@ def spin(self,conditions):
+        use_2d_analysis       = self.use_2d_analysis
+        rotation              = self.rotation           
+        pitch_c               = self.inputs.pitch_command  
+        
+        # Check for variable pitch
+        if np.any(pitch_c !=0) and not self.variable_pitch:
+            print("Warning: pitch commanded for a fixed-pitch rotor. Changing to variable pitch rotor for weights analysis.")
+            self.variable_pitch = True
+        
+        # Unpack freestream conditions
         rho     = conditions.freestream.density[:,0,None]
         mu      = conditions.freestream.dynamic_viscosity[:,0,None]
-        Vv      = conditions.frames.inertial.velocity_vector 
         a       = conditions.freestream.speed_of_sound[:,0,None]
         T       = conditions.freestream.temperature[:,0,None]
-        Na      = self.number_azimuthal_stations 
+        Vv      = conditions.frames.inertial.velocity_vector 
+        nu      = mu/rho   
+        rho_0   = rho
+        
+        # Helpful shorthands
         BB      = B*B    
         BBB     = BB*B
-    
-        # calculate total blade pitch
+        pi      = np.pi        
+        pi2     = pi*pi         
+     
+        # Calculate total blade pitch
         total_blade_pitch = beta_0 + pitch_c
     
-        # Velocities in the frame of the propeller
+        # Velocity in the rotor frame
         T_body2inertial = conditions.frames.body.transform_to_inertial
         T_inertial2body = orientation_transpose(T_body2inertial)
         V_body          = orientation_product(T_inertial2body,Vv)
         body2thrust     = self.body_to_prop_vel()
         T_body2thrust   = orientation_transpose(np.ones_like(T_body2inertial[:])*body2thrust)
         V_thrust        = orientation_product(T_body2thrust,V_body) 
-    
+        
+        # Check and correct for hover
         V         = V_thrust[:,0,None]  
         V[V==0.0] = 1E-6
-        ut        = np.zeros_like(V) 
-    
+        
         #Things that don't change with iteration
         Nr       = len(c) # Number of stations radially    
         ctrl_pts = len(Vv)
                  
-        # set up non dimensional radial distribution  
-        chi            = self.radius_distribution/R 
-        
-        omega          = np.abs(omega)        
-        r              = chi*R          # Radial coordinate 
-        pi             = np.pi        
-        pi2            = pi*pi       
-        n              = omega/(2.*pi)  # Cycles per second  
-        nu             = mu/rho         
+        # Non-dimensional radial distribution and differential radius
+        chi           = r_1d/R 
+        diff_r        = np.diff(r_1d)
+        deltar        = np.zeros(len(r_1d))
+        deltar[1:-1]  = diff_r[0:-1]/2 + diff_r[1:]/2
+        deltar[0]     = diff_r[0]/2
+        deltar[-1]    = diff_r[-1]/2
+        
+        # Check for negative rotation rate
+        if np.any(omega) <0:
```

Comment:
Does this really make sense?

Reply:
When I got to the bottom I understood the assumption you made. But it's still a little weird.

Reply:
probably not needed; I think we had this in an older version because there were some errors occurring at one point. I will remove the correction for negative rate since it shouldn't ever happen anyways